packages/teko/src/Teko_Utilities.cpp

```diff
@@ -238,11 +238,17 @@ RCP<Tpetra::CrsMatrix<ST,LO,GO,NT> > buildGraphLaplacian(int dim,ST * coords,con
    Teuchos::Array<ST> rowData(sz);
    Teuchos::Array<GO> rowInd(sz);
 
+#ifdef HAVE_TPETRA_INT_UNSIGNED_LONG
+   long diagInd;
```

Comment:
@ikalash You can do this without macros with `std::make_signed`.

Reply:
Thanks @mhoemmen I was going to suggest writing this, its nice to know the standard is somewhat sane.

Reply:
I re-did this w/o ifdefs - sorry about the confusion!  Here is the current commit showing the change: https://github.com/trilinos/Trilinos/pull/6454/commits/b8456c949203728608559390b59bebdca91e116d .

Reply:
@eric-c-cyr if it's not sane, please feel encouraged to write a paper and one of our C++ Standard Committee representatives will take it up at the next meeting :-)